### PR TITLE
(Slightly) Improve resolver performance

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* (Slightly) Improve resolver performance  
  [segiddins](https://github.com/segiddins)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/installer/analyzer/target_inspection_result.rb [hunk 1]
```diff
@@ -30,6 +30,10 @@ class TargetInspectionResult
         #
         attr_reader :archs
 
+        # @return [Pathname] the path to the root of the project containing the user target
+        #
+        attr_reader :client_root
```
Comment (dnkoutso) [inline lib/cocoapods/installer/analyzer/target_inspection_result.rb]: does this need to be exposed?

Comment (segiddins) [reply lib/cocoapods/installer/analyzer/target_inspection_result.rb]: It's used in the analyzer

**Comment by segiddins** [comment]
Rebased

**Pull request merged**